### PR TITLE
(helm) Support exposing additional ports for the Grafana Agent

### PR DESCRIPTION
**Pull request by payparain**
#### PR Description
This PR adds support for specifying extra port maps to the Grafana Agent, to support exposing different ports for specific components (such as [`otelcol.receiver.jaeger`](https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.receiver.jaeger/#thrift_http-block), which defaults to listening on `14268` for http-thrift traces). By supplying the values:

```
agent:
    extraPorts:
        - name: jaeger-thrift
           port: 14268
           targetPort: 14268
           protocol: TCP
```

A new port will be exposed in both the headless service and the container itself with the provided arguments, allowing one to use the Kube DNS to send otelcol data to the Agent.

#### Which issue(s) this PR fixes
N/A

#### Notes to the Reviewer
This approach is based on the way [`kube-prom-stack`](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/values.yaml#L359) adds additional ports for it's various components, but modified slightly due to the differences in how ports are specified on a container vs a service.
#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by payparain** [comment]
I hope I understood the tests right; I just put the rendered output into the relevant folder and added values like the others. Apologies if I got this wrong!

File: operations/helm/charts/grafana-agent/templates/containers/_agent.yaml
Comment (rfratto) [inline operations/helm/charts/grafana-agent/templates/containers/_agent.yaml:29]: Should this be targetPort instead of port? (since port refers to the service port and targetPort refers to the port number from the container)

File: operations/helm/charts/grafana-agent/Chart.yaml
Comment (rfratto) [inline operations/helm/charts/grafana-agent/Chart.yaml:5]: Can you bump this up to 0.2.0 instead since it's technically not fixing a bug?

File: operations/helm/charts/grafana-agent/CHANGELOG.md
Comment (rfratto) [inline operations/helm/charts/grafana-agent/CHANGELOG.md:10]: Looks like we'll have to release this immediately, can you add a section for 0.2.0?

We'll probably disable requiring version bumps for every change after this gets merged.

**Review by rfratto — APPROVED** [review]
Thank you! I'll merge once the feedback is addressed :)

File: operations/helm/charts/grafana-agent/templates/containers/_agent.yaml
Comment (payparain) [reply operations/helm/charts/grafana-agent/templates/containers/_agent.yaml:29]: Totally right! I'll change this now.

File: operations/helm/charts/grafana-agent/Chart.yaml
Comment (payparain) [reply operations/helm/charts/grafana-agent/Chart.yaml:5]: Yep, will do; wasn't sure what version to make it when bumping.

File: operations/helm/charts/grafana-agent/CHANGELOG.md
Comment (payparain) [reply operations/helm/charts/grafana-agent/CHANGELOG.md:10]: Will do

**Pull request merged**
